packages/path_provider/path_provider_platform_interface/lib/path_provider_platform_interface.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io' show Directory;
```

Comment:
I'm torn about making 'dart:io' part of the dependencies of the platform_interface. This makes it 100% non-implementable in web at all.

Any chance this platform interface can return paths (String) instead of fully-formed Directories?

Reply:
How about I rename those to getXPath instead of getXDirectory, and return the strings. The core plugin would actually make the `Directory` object?

Reply:
I just did that. I think it makes sense because it's now actually returning paths, not directories, and the plugin is called PATH_provider :) WDYT?

Reply:
Yep, the current version is way more portable! That way if we ever need to refactor this, we only need to go to the core plugin and change stuff there.